kubernetes/resource_kubernetes_job.go

```diff
@@ -111,6 +111,14 @@ func resourceKubernetesJobUpdate(ctx context.Context, d *schema.ResourceData, me
 
 	ops := patchMetadata("metadata.0.", "/metadata/", d)
 
+	if d.HasChange("spec") {
+		specOps, err := patchJobSpec("/spec", "spec.0.", d)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		ops = append(ops, specOps...)
+	}
+
```

Comment:
I pulled in Patrick's changes from here https://github.com/hashicorp/terraform-provider-kubernetes/pull/769 (This came over during my last bug on-duty, since I was reviewing high priority PRs).